purchase_add_product_supplierinfo/purchase.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def _check_product_supplierinfo(self):
+        lines = []
+        for line in self.order_line:
+            suppinfo = self.env['product.supplierinfo'].search([
+                ('product_id', '=', line.product_id.id),
+                ('name', '=', self.partner_id.id)])
+            if not suppinfo:
+                lines.append((0, 0, {
+                    'name': line.name,
+                    'product_id': line.product_id.id,
+                }))
+        return lines
+
+    @api.multi
+    def purchase_confirm(self):
+        self.ensure_one()
+        lines_for_update = self._check_product_supplierinfo()
+        if lines_for_update:
+            if self.partner_id.parent_id:
```

Comment:
I think there is a field or a method to get the commercial entity id. try self.partner_id._commercial_partner_id